File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    n = len(prices)
    stack, answer = [], [0] * n
    for i, price in enumerate(prices):
        while stack and stack[-1][1] > price:
            idx, p = stack.pop()
            answer[idx] = i - idx
        stack.append([i, price])
        
    while stack:
        idx, p = stack.pop()
        answer[idx] = n - idx - 1
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;

class Solution {
    public int[] solution(int[] prices) {
        int n = prices.length;
        
        int[] answer = new int[n];
        Deque<int[]> stack = new ArrayDeque<>();
        for (int i =  0; i < n; i++) {
            while (!stack.isEmpty() && stack.peek()[1] > prices[i]) {
                int[] temp = stack.pop();
                answer[temp[0]] = i - temp[0];
            }
            stack.push(new int[]{i, prices[i]});
        }
        
        while (!stack.isEmpty()) {
            int[] temp = stack.pop();
            answer[temp[0]] = n - temp[0] - 1;
        }
        return answer;
    }
}